src/main/java/com/microsoft/sqlserver/jdbc/KeyVaultCredential.java

```diff
@@ -37,11 +55,19 @@
     }
 
     public String doAuthenticate(String authorization, String resource, String scope) {
-        String accessToken;
+        String accessToken = null;
         if (null == authenticationCallback) {
-            AuthenticationResult token = getAccessTokenFromClientCredentials(authorization, resource, clientId,
-                    clientKey);
-            accessToken = token.getAccessToken();
+            if (null != msiCred) {
+                try {
+                    accessToken = msiCred.getToken(resource);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
```

Comment:
Why are we wrapping the exception into a RuntimException here? Shouldn't we wrap the exception into a SQLServerException instead?

Reply:
should we? I used RuntimeException to be consistent with what we're currently doing already.  getAccessTokenFromClientCredentials throws RuntimeException which just gets passed up to the caller here

Reply:
ok, I see that it's being done in the other method as well...but I think we usually wrap any exception into SQLServerException. But I'm ok with following the previous code.